File: SantoAndreOnBus.Api/Business/Lines/Line.cs
using SantoAndreOnBus.Api.Business.Places;
using SantoAndreOnBus.Api.Business.Vehicles;

namespace SantoAndreOnBus.Api.Business.Lines;

public record Line
{
    public int Id { get; set; }
    public string Identification { get; set; } = null!;
    public string? Fromwards { get; set; } = null!;
    public string? Towards { get; set; } = null!;
    public int DeparturesPerDay { get; set; }
    public virtual List<Place> Places { get; set; } = [];
    public virtual List<Vehicle> Vehicles { get; set; } = [];
}

File: BusCatalog.Spa/src/lines/line.ts
import { FormGroup } from "@angular/forms";
import { LineForm } from "./form/line-form.interface";

export class Line {
  constructor(
    public identification: string,
    public fromwards: string,
    public towards: string,
    public departuresPerDay: number,
    public vehicles: string[],
    public id?: number) {}

    static fromForm(form: FormGroup<LineForm>) {
      return new Line(
        form.get('identification')?.value!,
        form.get('fromwards')?.value!,
        form.get('towards')?.value!,
        form.get('departuresPerDay')?.value!,
        form.get('vehicles')?.value!);
    }
}